File: src/www/__test__/matchers/there_is_a_message_telling_X.js
import { getByTestId, getByText } from "@testing-library/dom";
import AbstractMatcher from "./AbstractMatcher";

export default class extends AbstractMatcher {
  constructor() {
    super(/there is a message telling _([^_]+)_/i);
  }

  interpretMatch([, text], _, { container, set }) {
    const messages = getByTestId(container, "messages");
    const message = getByText(messages, text);
    set("thatMessage", message);
  }
}

File: src/test/java/com/drpicox/game/matchers/map/There_is_a_message_telling_X.java
package com.drpicox.game.matchers.map;

import com.drpicox.game.interpreter.Context;
import com.drpicox.game.interpreter.Instruction;
import com.drpicox.game.tools.Json;
import org.springframework.stereotype.Component;

import java.util.regex.MatchResult;

import static com.drpicox.game.tools.JsonSubject.assertThat;

@Component
public class There_is_a_message_telling_X extends MapInstructionMatcher {
    public There_is_a_message_telling_X() {
        super("There is a message telling _([^_]+)_");
    }

    @Override
    public void interpretMatch(MatchResult match, Instruction instruction, Context context) {
        var game = context.wantJson("game");
        var expected = new Json("{}");
        expected.set("text", match.group(1));

        assertThat(game).at("messages").containsMatch(expected);

        var thatMessage = game.get("messages").findMatch(expected);
        context.set("thatMessage", thatMessage);
    }
}
